updater/site/github.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import re
+import random
+from updater.site.abstract_site import AbstractSite
+
+
+class Github(AbstractSite):
+    _URL = 'https://github.com/'
+
+    def __init__(self, url: str):
+        super().__init__(url)
+
+    @classmethod
+    def get_supported_urls(cls):
+        return [cls._URL]
+
+    def find_zip_url(self):
+        return self.url.replace('/tree/', '/archive/', 1) + '.zip'
+
+    def get_latest_version(self):
+        try:
+            response = requests.get(self.url)
+            response.raise_for_status()
+            content = str(response.content)
+            match = re.search(
+                r'<a data-pjax.*?\/commit\/(?P<hash>.*?)">',
+                content)
+            result = ''
+            if match:
```

Comment:
Nice! This can all be cleaned up a little bit, like how I am using regex for Curse:
```
version = re.search(
    r"your-regex",
    content_string).group('version')
return version
```

it will raise an exception if it fails to find the group